core/src/test/java/org/elasticsearch/test/ESBackcompatTestCase.java

```diff
@@ -141,6 +143,20 @@ private static Path backwardsCompatibilityPath() {
         return builder;
     }
 
+    @Override
+    public Settings indexSettings() {
+        // IndexMetaData.SETTING_DATA_PATH looks like it breaks backwards compatibility tests
```

Comment:
I think this is one of the things that I'd like to fix in https://github.com/elastic/elasticsearch/issues/13522